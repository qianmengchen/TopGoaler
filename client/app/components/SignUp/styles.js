import { StyleSheet } from 'react-native';

const card = StyleSheet.create({
  container: {
    backgroundColor: '#bbb',
    justifyContent: 'center',
    paddingTop: 22,
    flex: 1,
    flexDirection: 'column'
  },
  icons: {
    flexDirection: 'row',
    justifyContent: 'center'
  },
  button: {
    marginHorizontal: 40,
    marginTop: 10,
    paddingVertical: 10,
    backgroundColor: 'grey',
    borderRadius: 20,
    borderWidth: 1,
    borderColor: '#fff'
  },
  buttonText: {
    color: '#fff',
    textAlign: 'center',
    fontSize: 20
  },
  divider: {
    backgroundColor: 'grey',
    height: 1,
    flex: 1,
    alignSelf: 'center'
  },
  dividerText: {
    alignSelf: 'center',
    paddingHorizontal: 5,
    paddingVertical: 10,
    fontSize: 20,
    color: 'grey'
  },
  title: {
    alignSelf: 'center',
    fontSize: 40
  }
});

const alt = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignSelf: 'center'
  },
  text: {
    alignSelf: 'center',
    paddingTop: 20,
    color: 'grey'
  },
  button: {
    marginHorizontal: 40,
    marginTop: 10,
    paddingVertical: 10,
    borderRadius: 10
  },
  buttonText: {
    color: '#fff',
    textAlign: 'center',
    fontSize: 20,
    paddingLeft: 10
  },
  facebook: {
    backgroundColor: '#3b5998'
  },
  google: {
    backgroundColor: '#dd4b39'
  },
  twitter: {
    backgroundColor: '#00aced'
  }
});

export { card, alt };
