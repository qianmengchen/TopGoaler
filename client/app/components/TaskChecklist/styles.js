import { StyleSheet } from 'react-native';

const checklist = StyleSheet.create({
  container: {
    padding: 25
  },
  text: {
    fontSize: 20,
    color: 'white'
  }
});

const taskComponent = StyleSheet.create({
  container: {
    marginVertical: 15,
    flexDirection: 'row'
  },
  checkbox: {
    flex: 0.1,
    justifyContent: 'center',
    alignItems: 'center'
  },
  desc: {
    flex: 0.7,
    paddingLeft: 20,
    flexDirection: 'row'
  },
  point: {
    flex: 0.2,
    flexDirection: 'row',
    paddingRight: 10,
    justifyContent: 'flex-end'
  }
});

const checkbox = StyleSheet.create({
  circle: {
    width: 20,
    height: 20,
    borderRadius: 10,
    borderColor: 'white',
    borderWidth: 2
  }
});

export { checklist, taskComponent, checkbox };
