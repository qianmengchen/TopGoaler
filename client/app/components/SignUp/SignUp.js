import React, { Component } from 'react';
import { Text, View, TouchableHighlight } from 'react-native';
import { Card, Input, Icon } from 'react-native-elements';
import { card, alt } from './styles';

class SignUp extends Component {
  _submit() {
    this.props.submit(this.state.username, this.state.password);
    this.props.navigation.navigate('signIn');
  }
  _signIn() {
    this.props.navigation.navigate('signIn');
  }

  /**
   * This is the component that allows new users to signup to the application.
   * It will redirect users to signin page after signup succeeded.
   *
   * @class SignUp
   * @property {function} submit -  A function allowing for submitting the username and password to verify the credentials in our database.
   * @property {Object} navigation - An object that contains a navigate function allowing the user to navigate between components.
   */

  constructor(props) {
    super(props);
    this.state = {
      username: '',
      email: '',
      password: ''
    };
  }
  render() {
    return (
      <View style={card.container}>
        <Text style={card.title}>Top Goaler</Text>
        <Card title="Sign up for free">
          <Input
            containerStyle={{ marginBottom: 10 }}
            placeholder="Username"
            autoCapitalize="none"
            value={this.state.username}
            onChangeText={text => this.setState({ username: text })}
            leftIcon={<Icon name="person" size={24} color="grey" />}
          />
          <Input
            containerStyle={{ marginVertical: 10 }}
            placeholder="Password"
            autoCapitalize="none"
            textContentType="password"
            value={this.state.password}
            secureTextEntry={true}
            onChangeText={text => this.setState({ password: text })}
            leftIcon={<Icon name="lock" size={24} color="grey" />}
          />
          <Input
            containerStyle={{ marginVertical: 10 }}
            autoCapitalize="none"
            placeholder="Email"
            value={this.state.email}
            onChangeText={text => this.setState({ email: text })}
            leftIcon={<Icon name="email" size={24} color="grey" />}
          />
          <TouchableHighlight
            style={card.button}
            underlayColor="#aaa"
            onPress={this._submit.bind(this)}
          >
            <Text style={card.buttonText}> Create Account </Text>
          </TouchableHighlight>
        </Card>

        <Text style={alt.text}>Or you can also:</Text>
        <TouchableHighlight
          style={[alt.button, alt.facebook]}
          underlayColor="#aaa"
        >
          <View style={alt.container}>
            <Icon name="facebook-square" type="font-awesome" color="white" />
            <Text style={alt.buttonText}> Log in with Facebook </Text>
          </View>
        </TouchableHighlight>
        <TouchableHighlight
          style={[alt.button, alt.google]}
          underlayColor="#aaa"
        >
          <View style={alt.container}>
            <Icon name="google-plus" type="font-awesome" color="white" />
            <Text style={alt.buttonText}> Log in with Google </Text>
          </View>
        </TouchableHighlight>
        <TouchableHighlight
          style={[alt.button, alt.twitter]}
          underlayColor="#aaa"
        >
          <View style={alt.container}>
            <Icon name="twitter" type="font-awesome" color="white" />
            <Text style={alt.buttonText}> Log in with Twitter </Text>
          </View>
        </TouchableHighlight>

        <TouchableHighlight
          style={alt.button}
          underlayColor="#aaa"
          onPress={this._signIn.bind(this)}
        >
          <Text style={alt.buttonText}> I have an account </Text>
        </TouchableHighlight>
      </View>
    );
  }
}

export default SignUp;
