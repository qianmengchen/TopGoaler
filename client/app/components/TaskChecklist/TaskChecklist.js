import React, { Component } from 'react';
import { View, FlatList } from 'react-native';
import TaskEntry from './TaskEntry';
import * as styles from './styles';

class TaskChecklist extends Component {
  render() {
    return (
      <View style={styles.checklist.container}>
        <FlatList
          data={this.props.tasks}
          renderItem={TaskEntry}
          keyExtractor={item => item.id.toString()}
        />
      </View>
    );
  }
}

export default TaskChecklist;
