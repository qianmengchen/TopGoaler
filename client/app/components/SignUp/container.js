import { connect } from 'react-redux';
import SignUp from './SignUp';
import { navigateTo } from '../../actions';

const mapDispatchToProps = dispatch => ({
  reportNav: () => dispatch(navigateTo('TaskList'))
});

export default connect(
  () => ({}),
  mapDispatchToProps
)(SignUp);
