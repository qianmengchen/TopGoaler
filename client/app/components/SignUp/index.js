import SignUp from './container';

export { SignUp };
