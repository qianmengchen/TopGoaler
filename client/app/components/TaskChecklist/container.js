import { connect } from 'react-redux';
import TaskChecklist from './TaskChecklist';

const mapStateToProps = state => ({
  state: state
});

export default connect(mapStateToProps)(TaskChecklist);
