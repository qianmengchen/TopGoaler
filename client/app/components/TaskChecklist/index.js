import TaskChecklist from './container';

export { TaskChecklist };
