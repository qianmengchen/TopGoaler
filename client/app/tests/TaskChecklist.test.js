import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunkMiddleware from 'redux-thunk';
import sinon from 'sinon';
import { Input } from 'react-native-elements';
import renderer from 'react-test-renderer';
require('isomorphic-fetch');

import SignUp from '../components/SignUp/SignUp';
import TaskChecklist from '../components/TaskChecklist/TaskChecklist';

const middlewares = [thunkMiddleware]; // you can mock any middlewares here if necessary
const mockStore = configureStore(middlewares);

const initialTaskState = {
    tasks: ['do frontend', 'do backend', 'do MVP ftw'],
    isFetching: false,
    currentTaskPage: 'ChannelListPage'
  };

describe('Testing Task Checklist component', () => {
  const _submit = jest.fn();
  const navigation = { navigate: jest.fn() };
  const submit = sinon.spy(SignUp.prototype, '_submit');
  const signin = sinon.spy(SignUp.prototype, '_signIn');

  const wrapper = shallow(<TaskChecklist />, {
    context: { store: mockStore(initialTaskState) }
  });
  const render = wrapper.dive();

  it('renders as expected', () => {
    expect(render).toMatchSnapshot();
  });

  it('should contain a FlatList', () => {
    expect(render.find('FlatList')).toHaveLength(1);
  });
});
