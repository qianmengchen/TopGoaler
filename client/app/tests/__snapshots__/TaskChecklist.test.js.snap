// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Task Checklist component renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Component
    style={
      Object {
        "padding": 25,
      }
    }
  >
    <FlatList
      data={
        Array [
          Object {
            "key": "1",
            "name": "Leetcode Daily",
            "point": 30,
          },
          Object {
            "key": "2",
            "name": "Leetcode Daily",
            "point": 40,
          },
          Object {
            "key": "3",
            "name": "Leetcode Daily",
            "point": 50,
          },
          Object {
            "key": "4",
            "name": "Leetcode Daily",
            "point": 60,
          },
          Object {
            "key": "5",
            "name": "Leetcode Daily",
            "point": 70,
          },
          Object {
            "key": "6",
            "name": "Leetcode Daily",
            "point": 80,
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      renderItem={[Function]}
      scrollEventThrottle={50}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </Component>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <FlatList
        data={
          Array [
            Object {
              "key": "1",
              "name": "Leetcode Daily",
              "point": 30,
            },
            Object {
              "key": "2",
              "name": "Leetcode Daily",
              "point": 40,
            },
            Object {
              "key": "3",
              "name": "Leetcode Daily",
              "point": 50,
            },
            Object {
              "key": "4",
              "name": "Leetcode Daily",
              "point": 60,
            },
            Object {
              "key": "5",
              "name": "Leetcode Daily",
              "point": 70,
            },
            Object {
              "key": "6",
              "name": "Leetcode Daily",
              "point": 80,
            },
          ]
        }
        disableVirtualization={false}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onEndReachedThreshold={2}
        renderItem={[Function]}
        scrollEventThrottle={50}
        updateCellsBatchingPeriod={50}
        windowSize={21}
      />,
      "style": Object {
        "padding": 25,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "data": Array [
          Object {
            "key": "1",
            "name": "Leetcode Daily",
            "point": 30,
          },
          Object {
            "key": "2",
            "name": "Leetcode Daily",
            "point": 40,
          },
          Object {
            "key": "3",
            "name": "Leetcode Daily",
            "point": 50,
          },
          Object {
            "key": "4",
            "name": "Leetcode Daily",
            "point": 60,
          },
          Object {
            "key": "5",
            "name": "Leetcode Daily",
            "point": 70,
          },
          Object {
            "key": "6",
            "name": "Leetcode Daily",
            "point": 80,
          },
        ],
        "disableVirtualization": false,
        "horizontal": false,
        "initialNumToRender": 10,
        "keyExtractor": [Function],
        "maxToRenderPerBatch": 10,
        "numColumns": 1,
        "onEndReachedThreshold": 2,
        "renderItem": [Function],
        "scrollEventThrottle": 50,
        "updateCellsBatchingPeriod": 50,
        "windowSize": 21,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "View",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <FlatList
          data={
            Array [
              Object {
                "key": "1",
                "name": "Leetcode Daily",
                "point": 30,
              },
              Object {
                "key": "2",
                "name": "Leetcode Daily",
                "point": 40,
              },
              Object {
                "key": "3",
                "name": "Leetcode Daily",
                "point": 50,
              },
              Object {
                "key": "4",
                "name": "Leetcode Daily",
                "point": 60,
              },
              Object {
                "key": "5",
                "name": "Leetcode Daily",
                "point": 70,
              },
              Object {
                "key": "6",
                "name": "Leetcode Daily",
                "point": 80,
              },
            ]
          }
          disableVirtualization={false}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={1}
          onEndReachedThreshold={2}
          renderItem={[Function]}
          scrollEventThrottle={50}
          updateCellsBatchingPeriod={50}
          windowSize={21}
        />,
        "style": Object {
          "padding": 25,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "data": Array [
            Object {
              "key": "1",
              "name": "Leetcode Daily",
              "point": 30,
            },
            Object {
              "key": "2",
              "name": "Leetcode Daily",
              "point": 40,
            },
            Object {
              "key": "3",
              "name": "Leetcode Daily",
              "point": 50,
            },
            Object {
              "key": "4",
              "name": "Leetcode Daily",
              "point": 60,
            },
            Object {
              "key": "5",
              "name": "Leetcode Daily",
              "point": 70,
            },
            Object {
              "key": "6",
              "name": "Leetcode Daily",
              "point": 80,
            },
          ],
          "disableVirtualization": false,
          "horizontal": false,
          "initialNumToRender": 10,
          "keyExtractor": [Function],
          "maxToRenderPerBatch": 10,
          "numColumns": 1,
          "onEndReachedThreshold": 2,
          "renderItem": [Function],
          "scrollEventThrottle": 50,
          "updateCellsBatchingPeriod": 50,
          "windowSize": 21,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "View",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
